conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/RoutingRegistry.java

```diff
@@ -27,11 +27,11 @@
  */
 public interface RoutingRegistry {
 
-    RoutingRegistry get(String template, HttpHandler handler);
+    RoutingRegistry get(String template, String serviceName, String endpointName, HttpHandler handler);
 
-    RoutingRegistry post(String template, HttpHandler handler);
+    RoutingRegistry post(String template, String serviceName, String endpointName, HttpHandler handler);
 
-    RoutingRegistry put(String template, HttpHandler handler);
+    RoutingRegistry put(String template, String serviceName, String endpointName, HttpHandler handler);
 
-    RoutingRegistry delete(String template, HttpHandler handler);
+    RoutingRegistry delete(String template, String serviceName, String endpointName, HttpHandler handler);
```

Comment:
This breaks the audit library -- is that okay?

If we're going to break API I'd prefer to simplify RoutingRegistry to something along these lines to allow extensibility in the future:

```java
interface RoutingRegistry {
  RoutingRegistry register(EndpointRegistration);
}

final class EndpointRegistration {
   HttpHandler handler;
   pathTemplate; // e.g. /foo/{bar}
   requestMethod // e.g. GET
   optional<endpointName>
   optional<serviceName>
  ..etc..
}
```